### PR TITLE
trigger label colormap generation on seed change to fix shuffle bug, addresses #2523

### LINKED ISSUE #2523: Shuffle in labels layer broken in current master
## 🐛 Bug

In the current master, clicking shuffle in the labels layer has no effect.

## To Reproduce

1. Open the viewer with a label layer. E.g. run `examples/labels-2d.py`
2. Select the labels layer
3. Click the suffle button

## Expected behavior

Expected behaviour: label colours should change.

## Environment
napari: 0.4.8.dev22+g8d77f01d
Platform: Linux-5.8.0-48-generic-x86_64-with-debian-bullseye-sid
System: Ubuntu 20.04.2 LTS
Python: 3.7.9 (default, Aug 31 2020, 12:42:55) [GCC 7.3.0]
Qt: 5.15.2
PyQt5: 5.15.2
NumPy: 1.19.5
SciPy: 1.6.0
Dask: 2021.01.1
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.0 NVIDIA 460.39
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- affinder: 0.1.dev13+g89050b4
- console: 0.0.2
- napping: 0.1.2
- scikit-image: 0.4.8.dev22+g8d77f01d
- svg: 0.1.4

### PR DESCRIPTION
**Pull request by VolkerH**
# Description

For discussion see #2523

## Type of change

- [x ] Bug-fix (non-breaking change which fixes an issue)

# References
closes #2523

# How has this been tested?
manually verified that shuffling is restored

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

For discussion: not sure whether it is clean style to trigger an update through causing an Exception somewhere else (debatable). 
Probably, there should be an automated test to see whether shuffle does what it should. I don't have time for this in the near future though. Feel free to add to this branch (or use a completely different approach).

@DragaDoncila  @jni @sofroniewn

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! But let's wait for @DragaDoncila and @jni to comment before merge. Thanks!!

**Comment by sofroniewn** [comment]
If they approve of approach, adding a test before merge would be great too!!

**Comment by DragaDoncila** [comment]
My bad on introducing this one! 

> not sure whether it is clean style to trigger an update through causing an Exception somewhere else (debatable)

Only suggestion is in `@seed.setter` we could do `self._all_vals = np.asarray([])` rather than `None`, which would allow us to continue just catching `IndexError`.

**Comment by VolkerH** [comment]
> Only suggestion is in `@seed.setter` we could do `self._all_vals = np.asarray([])` rather than `None`, which would allow us to continue just catching `IndexError`.

Thanks for the suggestion. Done.

**Comment by VolkerH** [comment]
Added an assertion which checks that `_all_vals` changes when seed is set to a different value in `test_seed`.
This is a bit easier than testing that the shuffle button causes the change in the canvas (which I originally thought would be the test for #2523)

**Review by sofroniewn — COMMENTED** [review]
Changes and test look good. I will merge this now, thanks!!

**Pull request merged**
